File: connectors/connector-redmine/src/test/java/com/taskadapter/connector/redmine/GTaskToRedmineTest.java
package com.taskadapter.connector.redmine;

import com.taskadapter.connector.definition.exception.FieldConversionException;
import com.taskadapter.model.AllFields;
import com.taskadapter.model.GTask;
import com.taskadapter.redmineapi.bean.Issue;
import com.taskadapter.redmineapi.bean.Project;
import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Collections;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class GTaskToRedmineTest {
    private static Project project;

    @BeforeClass
    public static void beforeAll() {
        project = new Project(null);
    }

    @Test
    public void fieldsAreSet() throws FieldConversionException {
        var task = new GTask()
                .setValue(AllFields.summary, "summary")
                .setValue(AllFields.description, "descr");
        Issue issue = convert(task);
        assertThat(issue.getSubject()).isEqualTo("summary");
        assertThat(issue.getDescription()).isEqualTo("descr");
    }

    private Issue convert(GTask gTask) throws FieldConversionException {
        return getConverter().convert(gTask);
    }

    private GTaskToRedmine getConverter() {
        var config = new RedmineConfig();
        return new GTaskToRedmine(config,
                null, project, new RedmineUserCache(List.of()), Collections.emptyList(),
                Collections.emptyList(),
                Collections.emptyList(), Collections.emptyList());
    }


}

File: connectors/connector-redmine/src/test/scala/com/taskadapter/connector/redmine/GTaskToRedmineTest.scala
package com.taskadapter.connector.redmine

import java.util.Collections

import com.taskadapter.model.{GTask, Summary}
import com.taskadapter.redmineapi.bean.{Issue, ProjectFactory}
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class GTaskToRedmineTest extends FunSpec with Matchers {

  private val project = ProjectFactory.create()

  it("sets Summary") {
    val task = new GTask().setValue(Summary, "summary")
    convert(task).getSubject shouldBe "summary"
  }

  private def convert(gTask: GTask): Issue = getConverter().convert(gTask)

  private def getConverter(): GTaskToRedmine = {
    val config = new RedmineConfig()
    new GTaskToRedmine(config,
      null, project, new RedmineUserCache(Seq()), Collections.emptyList(),
      Collections.emptyList(),
      Collections.emptyList(), Collections.emptyList())
  }

}
